### PR TITLE
feat(vc-select): implement multiselect

### PR DESCRIPTION
**Pull request by goldenmaya**
### Jira-link:

https://virtocommerce.atlassian.net/browse/VCST-178
### Artifact URL:
https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.51.0-pr-939-8a07-8a07b901.zip

### CODE REVIEW

File: client-app/pages/account/checkout-defaults.vue [hunk 1]
```diff
@@ -34,26 +34,28 @@
           size="auto"
         >
           <template #placeholder>
-            <VcSelectItem>
-              <VcSelectItemImage src="/static/icons/placeholders/select-payment.svg" class="bg-gray-100/80" />
-              <VcSelectItemText>
-                {{ $t("common.placeholders.select_payment_method") }}
-              </VcSelectItemText>
-            </VcSelectItem>
+            <div class="flex items-center gap-3 p-3 text-sm">
```
Comment (asvishnyakov) [inline client-app/pages/account/checkout-defaults.vue:37]: Why do we moved from component to duplicated `div`'s with same classes? Looks like regression for me.
I think it's better to have component here, may be customizable. If we don't want to have specific for `VcSelect`, then reuse `MenuItem` or something like that.

File: client-app/ui-kit/components/molecules/select/vc-select.vue [hunk 2]
```diff
@@ -1,143 +1,125 @@
 <template>
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/molecules/select/vc-select.vue:1]: This component is too much complex, I think we need to find a way to separate it intomultiple components (single/multiple, container/filter/single/multi, etc.)

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: client-app/ui-kit/components/molecules/input/vc-input.vue [hunk 3]
```diff
@@ -144,6 +157,18 @@ function togglePasswordVisibility() {
   inputType.value = isPasswordVisible.value ? "text" : "password";
 }
 
+function handleContainerClick() {
+  if (inputElement.value) {
+    inputElement.value.focus();
+    inputElement.value.click();
+  }
+}
```
Comment (asvishnyakov) [inline client-app/ui-kit/components/molecules/input/vc-input.vue:165]: Why do we need this?

File: client-app/pages/account/checkout-defaults.vue (hunk 1 above)
Comment (goldenmaya) [reply client-app/pages/account/checkout-defaults.vue:37]: We have new design for selects https://virtocommerce.atlassian.net/browse/VCST-468

File: client-app/ui-kit/components/molecules/select/vc-select.vue (hunk 2 above)
Comment (goldenmaya) [reply client-app/ui-kit/components/molecules/select/vc-select.vue:1]: Yes, we can, we need to talk about it and create a new ticket.

File: client-app/ui-kit/components/molecules/input/vc-input.vue (hunk 3 above)
Comment (asvishnyakov) [reply client-app/ui-kit/components/molecules/input/vc-input.vue:165]: Needed to bypass events from inner elements like show password eye

File: client-app/ui-kit/components/molecules/select/vc-select.vue (hunk 2 above)
Comment (asvishnyakov) [reply client-app/ui-kit/components/molecules/select/vc-select.vue:1]: https://virtocommerce.atlassian.net/browse/VCST-471

**Review by asvishnyakov — APPROVED** [review]

**Review by NaMax66 — APPROVED** [review]

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**